test/e2e/helpers/machine.go

```diff
@@ -130,29 +132,42 @@ func CheckControlPlaneMachineOnDeleteReplacement(testFramework framework.Framewo
 // a replacement Machine.
 // It simultaneously checks that no other index is being replaced, this short circuits
 // what otherwise could be a long wait.
-func EventuallyIndexIsBeingReplaced(ctx context.Context, idx int) bool {
+func EventuallyIndexIsBeingReplaced(ctx context.Context, testFramework framework.Framework, idx int) bool {
 	controlPlaneMachineSelector := runtimeclient.MatchingLabels(framework.ControlPlaneMachineSetSelectorLabels())
+	k8sClient := testFramework.GetClient()
 
-	// Wait up to a minute for the replacement machine to be created.
-	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	// Wait for the replacement machine to be created.
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
 	defer cancel()
 
 	// Use RunCheckUntil to check that the other indexes do not get an additional
 	// replacement while we wait for the desired index to get a replacement.
 	passed := framework.RunCheckUntil(ctx,
 		func(_ context.Context, g framework.GomegaAssertions) bool { // Check function
 			By(fmt.Sprintf("Checking that other indexes (not %d) do not have 2 replicas", idx))
-			list := komega.ObjectList(&machinev1beta1.MachineList{}, controlPlaneMachineSelector)
 
-			return g.Expect(list()).Should(
+			list := &machinev1beta1.MachineList{}
+			if err := k8sClient.List(ctx, list, controlPlaneMachineSelector); err != nil {
+				// For temporary errors in listing objects we don't want to break this check,
+				// so we return happy and retry at the next check.
+				return g.Expect(err).Should(WithTransform(isRetryableAPIError(err), BeTrue()))
```

Comment:
Gomega should apply the transform if you pass a function in
```suggestion
				return g.Expect(err).Should(WithTransform(isRetryableAPIError, BeTrue()))
```

Reply:
👍